core/src/impl/Kokkos_Atomic_Load.hpp

```diff
@@ -168,6 +168,15 @@ inline T _atomic_load(T* ptr, MemoryOrder) {
   return *ptr;
 }
 
+#elif defined(KOKKOS_ENABLE_WINDOWS_ATOMICS)
+
+template <class T, class MemoryOrder>
+inline T _atomic_load(T* ptr, MemoryOrder) {
+  Kokkos::memory_fence();
+  return *ptr;
```

Comment:
This doesn’t make any sense. Probably you want to create a temporary andcreturn after second memory fence? But even then that is not correct for example for complex ...

Reply:
I am clearly out of my comfort zone here with `atomics` on Windows. If you have any good suggestions on what to do here, I am happy to adapt.

I mostly want to make sure that we can check that compiling with MSVC continues to work (via enabling appveyor for all PRs). From experience, it is a real struggle to keep that compiler happy if we don't check regularly. I would be fine with just aborting for cases that didn't work before and fix functionality and tests later.